src/FEValues/cell_values.jl

```diff
@@ -35,298 +35,96 @@ function default_geometric_interpolation(::Interpolation{shape}) where {dim, sha
-For more details see e.g. the doctoral thesis by Mirza Cenanovic, Ch. 2 **Tangential Calculus** [Cenanovic2017](@cite).
-"""
-embedding_det(J::SMatrix{3,2}) = norm(J[:,1] × J[:,2])
-
-"""
-    embedding_det(J::Union{SMatrix{2, 1}, SMatrix{3, 1}})
-
-Embedding determinant for curves in 2D and 3D.
-
-TLDR: "det(J) =" ||∂x/∂ξ||₂
-
-The transformation theorem for some function f on a 1D curve in 2D and 3D space leads to
-  ∫ f ⋅ dE = ∫ f ⋅ ∂x/∂ξ dξ = ∫ f ⋅ t ||∂x/∂ξ||₂ dξ
-where ||∂x/∂ξ||₂ is "detJ" and t is "the unit tangent".
-See e.g. https://scicomp.stackexchange.com/questions/41741/integration-of-d-1-dimensional-functions-on-finite-element-surfaces for simple explanation.
-"""
-embedding_det(J::Union{SMatrix{2, 1}, SMatrix{3, 1}}) = norm(J)
-# scalar interpolation with rdim < sdim
-function reinit!(cv::CellValues{<:Any, N_t, dNdx_t, dNdξ_t}, x::AbstractVector{Vec{sdim,T}}) where {
-    rdim, sdim, T,
-    N_t    <: Number,
-    dNdx_t <: SVector{sdim, T},
-    dNdξ_t <: SVector{rdim, T},
-}
-    _reinit_embedded!(cv, x, Val{sdim}(), Val{rdim}())
+@inline function reinit!(cv::CellValues, x::AbstractVector)
+    return reinit!(cv, nothing, x)
 end
-# vector interpolation with vdim != rdim and rdim < sdim
-function reinit!(cv::CellValues{<:Any, N_t, dNdx_t, dNdξ_t}, x::AbstractVector{Vec{sdim,T}}) where {
-    rdim, sdim, vdim, T,
-    N_t    <: SVector{vdim},
-    dNdx_t <: SMatrix{vdim, sdim, T},
-    dNdξ_t <: SMatrix{vdim, rdim, T},
-}
-    _reinit_embedded!(cv, x, Val{sdim}(), Val{rdim}())
-end
-# reinit! for embedded elements, rdim < sdim
-function _reinit_embedded!(cv::CellValues, x::AbstractVector{Vec{sdim,T}}, ::Val{sdim}, ::Val{rdim}) where {sdim, rdim, T}
-    @assert sdim > rdim "This reinit only works for embedded elements. Maybe you swapped the reference and spatial dimensions?"
-    n_geom_basefuncs = getngeobasefunctions(cv)
-    n_func_basefuncs = getnbasefunctions(cv)
-    length(x) == n_geom_basefuncs || throw_incompatible_coord_length(length(x), n_geom_basefuncs)
 
-    @inbounds for (i, w) in pairs(getweights(cv.qr))
-        fecv_J = zero(MMatrix{sdim, rdim, T}) # TODO replace with MixedTensor (see https://github.com/Ferrite-FEM/Tensors.jl/pull/188)
-        for j in 1:n_geom_basefuncs
-            #fecv_J += x[j] ⊗ cv.dMdξ[j, i] # TODO via Tensors.jl
-            for k in 1:sdim, l in 1:rdim
-                fecv_J[k, l] += x[j][k] * cv.dMdξ[j, i][l]
-            end
-        end
-        fecv_J = SMatrix(fecv_J)
-        detJ = embedding_det(fecv_J)
-        detJ > 0.0 || throw_detJ_not_pos(detJ)
-        cv.detJdV[i] = detJ * w
-        # Compute "left inverse" of J
-        Jinv = pinv(fecv_J)
-        for j in 1:n_func_basefuncs
-            #cv.dNdx[j, i] = cv.dNdξ[j, i] ⋅ Jinv # TODO via Tensors.jl
-            cv.dNdx[j, i] = dothelper(cv.dNdξ[j, i], Jinv)
-        end
+function reinit!(cv::CellValues, cell, x::AbstractVector{<:Vec})
```

Comment:
```suggestion
function reinit!(cv::CellValues, cell::Union{AbstractCell, Nothing}, x::AbstractVector{<:Vec})
```

Reply:
Fixed (also for FaceValues)
`PointValues` hasn't gotten this new interface yet, I'll make a reminder issue to support this after getting non-identity mapping interpolations.